pytx3/pytx3/indicator.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Wrappers for the json returned by the ThreatExchange API to typed objects.
+"""
+
+import typing as t
+
+from . import TE
+
+
+class ThreatIndicator(t.NamedTuple):
+    """
+    Wrapper around ThreatExchange JSON for a ThreatIndicator.
+
+    Example:
+    {
+      "id": "3058061737574159",
+      "indicator": "facefacefacefacefacefacefaceface",
+      "type": "HASH_PDQ",
+      "creation_time": 1601513795,
+      "last_updated": 1601513796,
+      "status": "MALICIOUS",
+      "is_expired": false,
+      "tags": [
+        "tag1",
+        "tag2",
+        "tag3"
+      ],
+      "applications_with_opinions": [
+        "123456789055502"
+      ],
+      "expire_time": 9601513796
+    }
+    """
+
+    id: int
+    indicator: str
+    threat_type: str
+    creation_time: int
+    last_updated: int
+    status: str
+    is_expired: bool
+    tags: t.List[str]
+    applications_with_opinions: t.List[int]
+    expire_time: int
+
+    # def __init__(
+    #     self,
+    #     id: int,
+    #     indicator: str,
+    #     threat_type: str,
+    #     creation_time: int,
+    #     last_updated: int,
+    #     status: str,
+    #     is_expired: bool,
+    #     tags: t.List[str],
+    #     applications_with_opinions: t.List[int],
+    #     expire_time: int,
+    # ):
+    #     self.id = id
+    #     self.indicator = indicator
+    #     self.threat_type = threat_type
+    #     self.creation_time = creation_time
+    #     self.last_updated = last_updated
+    #     self.status = status
+    #     self.is_expired = is_expired
+    #     self.tags = tags
+    #     self.applications_with_opinions = applications_with_opinions
+    #     self.expire_time = expire_time
+
+    def as_row(self) -> t.Tuple[int, str, str, int, int, str, bool, str, str, int]:
+        """Simple conversion to CSV row"""
+        return (
+            self.id,
+            self.indicator,
+            self.threat_type,
+            self.creation_time,
+            self.last_updated,
+            self.status,
+            self.is_expired,
+            " ".join(self.tags),
+            " ".join([str(app) for app in self.applications_with_opinions]),
```

Comment:
Why do we join these and then later split them. Why not keep nested structure?

Reply:
Following the format used in descriptor.py.

Alternatively I can store the list directly and it would be saved into the file like "[123,456,789]".

Reply:
Since this is for translation to the CSV format and is the same approach used by SimpleDescriptorRollup.as_row in descriptor.py, I'm good with keeping the join method here. I'm not a fan of having to escape things in CSV, so if we can avoid having to escape the commas, I'm all for it.

Reply:
sgtm